project-minimization-plugin/src/test/kotlin/MinimizationPluginStateTest.kt

```diff
@@ -1,53 +1,110 @@
-import com.intellij.openapi.util.JDOMUtil
-import com.intellij.util.xmlb.XmlSerializer
-import junit.framework.TestCase
-import mu.KotlinLogging
+import com.intellij.openapi.components.service
+import com.intellij.testFramework.fixtures.BasePlatformTestCase
 import org.plan.research.minimization.plugin.model.FileLevelStage
-import org.plan.research.minimization.plugin.model.state.CompilationStrategy
-import org.plan.research.minimization.plugin.model.state.DDStrategy
+import org.plan.research.minimization.plugin.model.FunctionLevelStage
+import org.plan.research.minimization.plugin.model.state.*
+import org.plan.research.minimization.plugin.services.MinimizationPluginSettings
 import org.plan.research.minimization.plugin.settings.MinimizationPluginState
+import org.plan.research.minimization.plugin.settings.loadStateFromFile
+import org.plan.research.minimization.plugin.settings.saveStateToFile
+import org.w3c.dom.Document
+import java.io.File
+import javax.xml.parsers.DocumentBuilderFactory
+import kotlin.test.assertEquals
 
-class MinimizationPluginStateTest : TestCase() {
+class MinimizationPluginStateTest : BasePlatformTestCase() {
 
-    fun testDefault() {
-        val state = MinimizationPluginState()
-        doTest(state)
-    }
-    
-    fun testModified() {
-        val state = MinimizationPluginState()
-        state.compilationStrategy = CompilationStrategy.DUMB
-        state.gradleTask = "compileKotlin"
-        state.temporaryProjectLocation = "custom-location"
-        state.gradleOptions = listOf("--option1", "--option2")
-        state.stages = listOf(FileLevelStage())
-        state.minimizationTransformations = listOf()
-        doTest(state)
-    }
+    override fun setUp() {
+        super.setUp()
+        project.service<MinimizationPluginSettings>().updateState(MinimizationPluginState())
 
-    fun testStages() {
-        val state = MinimizationPluginState()
-        state.stages = listOf(
+        saveStateToFile(project, "baseState.xml")
+
+        val newState = MinimizationPluginState()
+        newState.compilationStrategy = CompilationStrategy.DUMB
+        newState.gradleTask = "user_build"
+        newState.gradleOptions = listOf("--info")
+        newState.temporaryProjectLocation = "new-project-location"
+        newState.snapshotStrategy = SnapshotStrategy.PROJECT_CLONING
+        newState.exceptionComparingStrategy = ExceptionComparingStrategy.SIMPLE
+        newState.stages = listOf(
             FileLevelStage(
-                ddAlgorithm = DDStrategy.DD_MIN
+                hierarchyCollectionStrategy = HierarchyCollectionStrategy.FILE_TREE,
+                ddAlgorithm = DDStrategy.DD_MIN,
             )
         )
-        doTest(state)
+        newState.minimizationTransformations = emptyList()
+
+        project.service<MinimizationPluginSettings>().updateState(newState)
+
+        saveStateToFile(project, "changedState.xml")
```

Comment:
This setup relies on the functionality that should be tested in the test case. It's better to create some files in resources manually